src/go/guestagent/pkg/tracker/apitracker.go

```diff
@@ -52,25 +44,29 @@ var (
 // and unexposing the ports on the host. This should only be used when
 // the Rancher Desktop networking is enabled and the privileged service is disabled.
 type APITracker struct {
-	forwarder      forwarder.Forwarder
-	isAdmin        bool
-	baseURL        string
-	tapInterfaceIP string
-	httpClient     http.Client
-	portStorage    *portStorage
+	context           context.Context
+	wslProxyForwarder forwarder.Forwarder
+	isAdmin           bool
+	enableListeners   bool
+	baseURL           string
+	tapInterfaceIP    string
+	portStorage       *portStorage
+	apiForwarder      *forwarder.APIForwarder
```

Comment:
If I understand correctly, `apiForwarder` is only used when `enableListeners` is set? Would it make sense then to just make the nil-ness of `apiForwarder` be the boolean?

Reply:
… I understand wrongly, we call `a.AddListener` instead.